src/request_handler.rs

```diff
@@ -24,54 +25,54 @@ pub fn global_request_entrypoint(
-    let body_future = req.into_body().concat2().map(|c| {
-        // Convert the Chunk into a rust "String", wrapping any error in our error type
-        str::from_utf8(&c).map(str::to_owned).map_err(WorkerError::from)
-    });
+    // Get a stream of Bytes representing the body of the request
+    let mut body_stream = req.into_body();
+    // Turn that stream into a concrete String
+    let mut body = String::new();
+    while let Some(chunk) = body_stream.next().await {
+        body.push_str(str::from_utf8(&chunk?)?);
+    }
 
-    // Next we want to an operation on the body. This needs to happen in a future for two reasons
-    // 1) We want to handle many requests at once, so we don't want to block a thread
-    // 2) Hyper literally doesn't let you deal with the body unless you're inside a future context (there is no API to escape this)
-    // Note: We already have a result (body_result) here, since we might get an Utf8 decode error above
-    body_future.map(move |body_result| {
-        debug!("body = {:?}", body_result);
-
-        let resp = body_result
-            // Delegate to the handler to actually deal with this request
-            .and_then(|body| {
-                // NOTE: We cannot handle panics here, since it could leave the handler in an inconsistent state
-                // Better to just bomb out
-                // TODO: Investigate handling panics at a lower level
-                handler.handle(http_verb, &uri, query, body)
-            })
-            .unwrap_or_else(|e| {
-                warn!("Forced to convert error {:?} into a http response", e);
-                e.into()
-            });
-
-        if resp.status() == StatusCode::INTERNAL_SERVER_ERROR {
-            error!("INTERNAL SERVER ERROR -- {:?}", resp);
-        } else {
-            debug!("{:?}", resp);
-        }
+    debug!("body = {:?}", body);
 
-        resp
+    // We want to do the actual handling in a "spawn_blocking" closure, since many operations there can block
+    // This allows us to handle a ton of requests at once, since we're not blocking the executor
+    let resp = spawn_blocking(move || {
+        // Delegate to the handler to actually deal with this request
+        // NOTE: We cannot handle panics here, since it could leave the handler in an inconsistent state
+        // Better to just bomb out
+        // TODO: Investigate handling panics at a lower level
+        handler.handle(http_verb, &uri, query, body)
     })
+    .await?
+    .unwrap_or_else(|e| {
+        warn!("Forced to convert error {:?} into a http response", e);
+        e.into()
+    });
+
+    if resp.status() == StatusCode::INTERNAL_SERVER_ERROR {
+        error!("INTERNAL SERVER ERROR -- {:?}", resp);
+    } else {
+        debug!("{:?}", resp);
+    }
+
+    Ok(resp)
 }
 
 #[derive(Debug)]
 pub struct HttpRequestHandler {
     serverless_component_manager: RwLock<ComponentManager>,
 }
 
+#[allow(clippy::unused_self)]
```

Comment:
Why'd you add this?

Reply:
Because technically the function could be static/not use the `self` parameter. Clippy catches this, but as a style point I like it this way (and that function could easily grow to use self).